libgalois/include/katana/PropertyGraph.h

```diff
@@ -842,6 +903,633 @@ CreateSymmetricGraph(PropertyGraph* pg);
+    return MakeStandardRange<edge_iterator>(edge_begin(node), edge_end(node));
+  }
+
+  /**
+   * @param N node to get edges for
+   * @param label label to get edges of
+   * @returns Range to edges of node N
+   */
+  edges_range edges(Node N, const EdgeLabelTy& label) const noexcept {
+    return katana::MakeStandardRange(edge_begin(N, label), edge_end(N, label));
+  }
+
+  Node edge_dest(Edge edge_id) const noexcept {
+    KATANA_LOG_DEBUG_ASSERT(edge_id < dests_.size());
+    return dests_[edge_id];
+  }
+
+  Node edge_dest(const edge_iterator ei) const noexcept {
+    return edge_dest(*ei);
+  }
+
+  /**
+   * @param node node to get degree for
+   * @returns Degree of node N
+   */
+  size_t degree(Node node) const noexcept {
+    auto edge_range = edges(node);
+    return std::distance(edge_range.begin(), edge_range.end());
+  }
+
+  /**
+   * @param N node to get degree for
+   * @param label label to get degree of
+   * @returns Degree of node N
+   */
+  size_t degree(Node N, const EdgeLabelTy& label) const noexcept {
+    return std::distance(edge_begin(N, label), edge_end(N, label));
+  }
+
+  nodes_range nodes(Node begin, Node end) const noexcept {
+    return MakeStandardRange<node_iterator>(begin, end);
+  }
+
+  nodes_range all_nodes() const noexcept {
+    return nodes(Node{0}, static_cast<Node>(num_nodes()));
+  }
+
+  edges_range all_edges() const noexcept {
+    return MakeStandardRange<edge_iterator>(Edge{0}, Edge{num_edges()});
+  }
+  // Standard container concepts
+
+  node_iterator begin() const noexcept { return node_iterator(0); }
+
+  node_iterator end() const noexcept { return node_iterator(num_nodes()); }
+
+  size_t size() const noexcept { return num_nodes(); }
+
+  bool empty() const noexcept { return num_nodes() == 0; }
+
+  Edge original_edge_id(const edge_iterator& e_iter) const noexcept {
+    return original_edge_id(*e_iter);
+  }
+
+  Edge original_edge_id(const Edge& e) const noexcept {
+    KATANA_LOG_DEBUG_ASSERT(e < num_edges());
+    return orig_edge_ids_[e];
+  }
+
+  Node original_node_id(const Node& nid) const noexcept { return nid; }
+
+  auto edge_label(Edge eid) const noexcept {
+    return prop_graph_->GetEdgeTypeSetID(original_edge_id(eid));
+  }
+
+  auto GetDistinctEdgeLabels() const noexcept {
+    return edge_label_index_->distinct_edge_labels();
+  }
+
+  bool DoesEdgeLabelExist(const EdgeLabelTy& label) const noexcept {
+    return edge_label_index_->has_edge_label(label);
+  }
+
+  struct EdgeComparator {
+    const EdgeLabelAwareTopology* outer_;
+
+    bool operator()(const edge_iterator& i, const Node& n) const noexcept {
+      return outer_->edge_dest(i) < n;
+    }
+
+    bool operator()(const Node& n, const edge_iterator& i) const noexcept {
+      return n < outer_->edge_dest(i);
+    }
+  };
+
+  /**
+   * Returns all edges from src to dst with some label.  If not found, returns
+   * nothing.
+   */
+  std::optional<edges_range> FindAllEdgesWithLabel(
```

Comment:
return empty range instead of `optional`?

Reply:
done.